ceph/rbd/workflows/snap_clone_operations.py

```diff
@@ -278,6 +278,21 @@ def clone_ops(**kw):
                 return 1
             clones = json.loads(out)
             for clone in clones:
+                # due to parallel support, have to check if clone already flattened or not
+                du_out, du_err = rbd.du(pool=clone["pool"], format="json")
+                if du_err:
+                    log.error(f"rbd du failed for {pool} with error {du_err}")
+                    if test_ops_parallely:
+                        raise Exception(f"rbd du failed for {pool} with error {du_err}")
+                    return 1
+                du_out_json = json.loads(du_out)
+                for _image in du_out_json["images"]:
+                    if _image["name"] == clone["image"]:
+                        if _image["used_size"] != 0:
```

Comment:
does used bytes depends on clone is flattened or not?

Reply:
yup, it will fail as image is already flattened since we run in parallel. The problem is "snap protect + clone + flatten + snap unprotect" is too tightly coupled so in parallel flatten gets picked up for same clone image multiple times, so added this safeguard instead of decoupling "snap protect + clone + flatten + snap unprotect"